install/tools/ipa-pki-wait-running.in

```diff
@@ -74,9 +74,13 @@ def get_status(conn, timeout):
     """
     client = SystemStatusClient(conn)
     response = client.get_status(timeout=timeout)
-    root = ElementTree.fromstring(response)
-    status = root.findtext("Status")
-    error = root.findtext("Error")
+    status = None
+    error = None
+    try:
+        json_response = json.loads(response)
+        status = json_response['Response']['Status']
```

Comment:
What happens if valid JSON is returned but these elements are not in it?

Reply:
`status` is `None` going into this block, so line 81 will `raise KeyError` and `status` will remain `None`.

I tried to adhere to the original intent of the code so as to not change behaviour. `status = root.findtext("Status")` returns `None` if the string is not found, so I made the new code do the same. Likewise for the error.

I can certainly catch more specific exceptions, `JSONDecodeError` from line 80 and `KeyError` for line 81. Done in next revision.